### PR TITLE
[Diag] NFC: Port diagnostic messages from `.def` files to `en.yaml`

### PR DESCRIPTION
**Pull request by HassanElDesouky**
In this PR I'm porting diagnostic messages from `.def` files. to the new `en.yaml` file. 

To test this, I parsed this YAML file and checked against all of the messages from `.def` files. And everything matched.

cc @xedin

### CODE REVIEW

File: lib/AST/LocalizationFormat.cpp [hunk 1]
```diff
@@ -101,6 +101,7 @@ readYAML(llvm::yaml::IO &io, T &Seq, bool, Context &Ctx) {
       DiagnosticNode current;
       yamlize(io, current, true, Ctx);
       io.postflightElement(SaveInfo);
+      std::replace(current.msg.begin(), current.msg.end(), '\n', ' ');
```
Comment (HassanElDesouky) [inline lib/AST/LocalizationFormat.cpp:104]: @xedin The parsed multiline YAML string will have line breaks `\n` in it. So, I had to manually replace them with spaces.

**Comment by HassanElDesouky** [comment]
@xedin I ran tests locally using `en` locale and tests passed. I also checked the content of the parsed strings against the content of the `diagnosticMessages` array (the one form `.def` files) and also everything matched.

File: lib/AST/LocalizationFormat.cpp (hunk 1 above)
Comment (theblixguy) [reply lib/AST/LocalizationFormat.cpp:104]: Should this also check for other kinds of line breaks, such as `\r\n` (on Windows)?

Comment (HassanElDesouky) [reply lib/AST/LocalizationFormat.cpp:104]: I don't have a Windows machine so I don't really know. How should I test it?

Comment (xedin) [reply lib/AST/LocalizationFormat.cpp:104]: Let's not do this at all for now, we can validate YAML format to detect such problems early.

**Review by xedin — APPROVED** [review]
Thank you!

**Comment by xedin** [comment]
@swift-ci please smoke test

**Pull request merged**